plutus-chain-index-core/src/Plutus/ChainIndex/Tx.hs

```diff
@@ -117,13 +118,21 @@ mintingPolicies = Map.fromList . fmap withHash . Set.toList
     withHash mp = let (MintingPolicyHash mph) = mintingPolicyHash mp
                    in (ScriptHash mph, getMintingPolicy mp)
 
-validators :: Set TxIn -> (Map ScriptHash Script, Map DatumHash Datum, Map RedeemerHash Redeemer)
-validators = foldMap (maybe mempty withHash . txInType) . Set.toList
+validators :: Set TxIn -> (Map ScriptHash Script, Map DatumHash Datum, Redeemers)
+validators txIns = foldMap (\(ix, txIn) -> maybe mempty (withHash ix) $ txInType txIn) $ zip [0..] (Set.toList txIns)
   where
-    withHash (ConsumeScriptAddress val red dat) =
+    -- TODO: the index of the txin is probably incorrect as we take it from the set.
```

Comment:
Is this going to have bad consequences?

Reply:
My understanding is that this PR makes a move from a completely broken state to a semi-broken state. Instead of not getting redeemers at all, we will start choosing them but with a big chance of selecting the incorrect ones.

Reply:
Speaking of `fromOnChainTx` in particular, I think `withHash` function was useless already since we use the `Tx` here, and the inputs have an empty `TxInType` here as I understand.

Reply:
>As long as the redeemer pointer points to the right input, I think this is totally fine.

There is a problem here that we take a set that could break the order. I looked at usages of the `fromOnChainTx` and there is no anything that directly relies on that order. But would be nice to fix `Tx` type.